src/session/session_api.c

```diff
@@ -713,20 +713,6 @@ __wt_open_cursor(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *owner, co
     hash_value = 0;
     WT_NOT_READ(txn_global, &S2C(session)->txn_global);
 
-    /*
```

Comment:
What misbehavior was this restriction protecting against? What changed so it is no longer necessary to protect against that misbehavior?

Reply:
My vaguely remember this is about a deadlock situation if we open a lot of history store. May be I'll hit the deadlock and make this change not feasible. But currently I'm still trying.